src/commandcollection.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module core/commandcollection
+ */
+
+import CKEditorError from '@ckeditor/ckeditor5-utils/src/ckeditorerror';
+
+/**
+ * Collection of commands. Its instance is available in {@module:core/editor/editor~Editor#commands}.
+ */
+export default class CommandCollection {
+	/**
+	 * Creates collection instance.
+	 */
+	constructor() {
+		/**
+		 * Command map.
+		 *
+		 * @private
+		 * @member {Map}
+		 */
+		this._commands = new Map();
+	}
+
+	/**
+	 * Registers a new command.
+	 *
+	 * @param {String} commandName The name of the command.
+	 * @param {module:core/command~Command} command
+	 */
+	add( commandName, command ) {
```

Comment:
One more thing – in `MarkerCollection` there's `set()` because it allows to override values (it even fires `remove` and `add`). We don't expect that in `CommandCollection` for instance – it should throw (it doesn't now, but that was the assumed semantic).